distributed/http/scheduler/tests/test_missing_bokeh.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+import sys
+from unittest import mock
+
+import pytest
+from packaging.version import Version
+from tornado.httpclient import AsyncHTTPClient
+
+from distributed import Scheduler
+from distributed.utils_test import gen_test
+
+
+@gen_test()
+async def test_missing_bokeh():
+    with mock.patch.dict(sys.modules, {"bokeh": None}):
+        # NOTE: mocking here is fiddly; if any of these get refactored or import order
+        # changes, this may break.
+        sys.modules.pop("distributed.dashboard.scheduler", None)
+        sys.modules.pop("distributed.dashboard.core", None)
+        sys.modules.pop("distributed.dashboard.utils", None)
+
+        async with Scheduler(dashboard_address=":0") as s:
+            http_client = AsyncHTTPClient()
+            response = await http_client.fetch(
+                f"http://localhost:{s.http_server.port}/status"
+            )
+            assert response.code == 200
+            body = response.body.decode()
+            assert "Dask needs bokeh" in body
+
+
+@gen_test()
+async def test_bokeh_version_too_low():
+    pytest.importorskip("bokeh")
+
+    with mock.patch.dict(sys.modules):
+        # Remove these imports, so when the scheduler imports
+        # `distributed.dashboard.scheduler`, it has to re-import `dashboard.core`, which
+        # is where the bokeh version detection happens at import time, via
+        # `dashboard.utils.BOKEH_VERSION`.
+        sys.modules.pop("distributed.dashboard.scheduler", None)
+        sys.modules.pop("distributed.dashboard.core", None)
```

Comment:
Agreed, it's highly brittle (as evidenced by it working the first time but failing when run in CI). Since all this logic happens at import time, brittle mocks are kind of the only option—we'd probably want to refactor things into a function, so patching it is much more predictable.